src/Compilers/CSharp/Test/Semantic/Utilities/ValueSetTests.cs

```diff
@@ -0,0 +1,529 @@
+            Assert.True(s1.All(LessThan, 2));
+            Assert.True(s1.All(LessThanOrEqual, 1));
+            Assert.False(s1.All(GreaterThan, 0));
+            Assert.False(s1.All(GreaterThanOrEqual, 0));
+            Assert.False(s1.All(Equal, 0));
+
+            Assert.False(s1.All(LessThan, -10));
+            Assert.False(s1.All(LessThanOrEqual, -10));
+            Assert.False(s1.All(GreaterThan, -10));
+            Assert.False(s1.All(GreaterThanOrEqual, -10));
+            Assert.False(s1.All(Equal, -10));
+            Assert.True(s1.All(LessThan, 10));
+            Assert.True(s1.All(LessThanOrEqual, 10));
+            Assert.False(s1.All(GreaterThan, 10));
+            Assert.False(s1.All(GreaterThanOrEqual, 10));
+            Assert.False(s1.All(Equal, 10));
+
+            var s2 = ForLong.Related(GreaterThan, -5).Intersect(s1);
+            Assert.Equal($"[-4..1]", s2.ToString());
+            Assert.False(s2.All(LessThan, -10));
+            Assert.False(s2.All(LessThanOrEqual, -10));
+            Assert.True(s2.All(GreaterThan, -10));
+            Assert.True(s2.All(GreaterThanOrEqual, -10));
+            Assert.False(s2.All(Equal, -10));
+            Assert.True(s2.All(LessThan, 10));
+            Assert.True(s2.All(LessThanOrEqual, 10));
+            Assert.False(s2.All(GreaterThan, 10));
+            Assert.False(s2.All(GreaterThanOrEqual, 10));
+            Assert.False(s2.All(Equal, 10));
+
+            IValueSet b = s1;
+            Assert.Same(s1.Factory, b.Factory);
+            Assert.Same(s1.Factory.None, b.Factory.None);
+        }
+
+        [Fact]
+        public void TestNext_Cov_01()
+        {
+            Assert.Equal("[10..100]", ForSByte.Related(GreaterThanOrEqual, 10).Intersect(ForSByte.Related(LessThanOrEqual, 100)).ToString());
+            Assert.Equal("[10..100]", ForShort.Related(GreaterThanOrEqual, 10).Intersect(ForShort.Related(LessThanOrEqual, 100)).ToString());
+            Assert.Equal("[10..100]", ForUInt.Related(GreaterThanOrEqual, 10).Intersect(ForUInt.Related(LessThanOrEqual, 100)).ToString());
+            Assert.Equal("[10..100]", ForULong.Related(GreaterThanOrEqual, 10).Intersect(ForULong.Related(LessThanOrEqual, 100)).ToString());
+            Assert.Equal("[10..100]", ForUShort.Related(GreaterThanOrEqual, 10).Intersect(ForUShort.Related(LessThanOrEqual, 100)).ToString());
+            Assert.Equal("[10..100]", ForFloat.Related(GreaterThanOrEqual, 10).Intersect(ForFloat.Related(LessThanOrEqual, 100)).ToString());
+            Assert.Equal("[-100..-10]", ForFloat.Related(GreaterThanOrEqual, -100).Intersect(ForFloat.Related(LessThanOrEqual, -10)).ToString());
+            Assert.Equal("[-10..10]", ForFloat.Related(GreaterThanOrEqual, -10).Intersect(ForFloat.Related(LessThanOrEqual, 10)).ToString());
+        }
+
+        [Fact]
+        public void TestAPI_01()
+        {
+            //Assert.Same(ForByte, ForSpecialType(SpecialType.System_Byte));
```

Comment:
>Assert [](start = 14, length = 6)

I need to uncomment these lines.  I commented them out to check if we had coverage even without this test (we do) but forgot to uncomment them.